client/app/bundles/course/container/CourseContainer.tsx

```diff
@@ -0,0 +1,71 @@
+import { ComponentRef, useRef, useState } from 'react';
+import { LoaderFunction, Outlet, useLoaderData } from 'react-router-dom';
+import { MenuOutlined } from '@mui/icons-material';
+import { IconButton } from '@mui/material';
+import { CourseLayoutData } from 'types/course/courses';
+
+import CourseAPI from 'api/course';
+import PopupNotifier from 'course/user-notification/PopupNotifier';
+import Footer from 'lib/components/core/layouts/Footer';
+import { DataHandle, useDynamicNest } from 'lib/hooks/router/dynamicNest';
+
+import Breadcrumbs from './Breadcrumbs';
+import Sidebar from './Sidebar';
+
+const CourseContainer = (): JSX.Element => {
+  const data = useLoaderData() as CourseLayoutData;
+
+  const sidebarRef = useRef<ComponentRef<typeof Sidebar>>(null);
+  const [sidebarOpen, setSidebarOpen] = useState(false);
+
+  const { crumbs, loading, activePath } = useDynamicNest();
+
+  return (
+    <main className="flex h-full min-h-0 w-full">
+      <Sidebar
+        ref={sidebarRef}
+        activePath={activePath}
+        from={data}
+        onChangeVisibility={setSidebarOpen}
+      />
+
+      <aside className="h-full w-full overflow-scroll">
+        <div className="flex h-[4rem] w-full items-center">
+          {!sidebarOpen && (
+            <IconButton onClick={(): void => sidebarRef.current?.show()}>
+              <MenuOutlined />
+            </IconButton>
+          )}
+
+          <Breadcrumbs
+            className="h-[4rem] w-full overflow-hidden"
+            in={crumbs}
+            loading={loading}
+          />
+        </div>
+
+        <div className="min-h-[calc(100%_-_4rem)] w-full">
+          <Outlet />
+        </div>
+
+        <Footer />
```

Comment:
In course pages, the sidebar need to take precedence. Adding `Footer` to `AppContainer` will make the footer appear in the top-most element, rather than the child that `CourseContainer` has designated.

The right way is to not add `Footer` to `AppContainer`, but add it to `CourseContainer` and `CourselessContainer`. Let these components decide where to place the footer for themselves.